content/contexts/experimenterDetails.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+
+interface ExperimenterData {
+  publicDescription: string;
+  proposedStartDate: Date;
+  proposedEndDate: Date;
+  proposedDuration: number;
+  startDate?: Date;
+  endDate?: Date;
+  variants: Record<string, string>;
+}
+
+interface ExperimenterState {
+  data: ExperimenterData;
+}
+
+interface ExperimenterState {
+  data: ExperimenterData;
+}
+
+const initialState: ExperimenterState = {
+  data: null,
+};
+
+export const experimenterDetailsContext = React.createContext(initialState);
+const { Provider } = experimenterDetailsContext;
+
+export const ACTION_UPDATE_DATA = "UPDATE_DATA";
+
+function reducer(state, action): ExperimenterState {
+  switch (action.type) {
+    case ACTION_UPDATE_DATA:
+      return {
+        ...state,
+        data: {
+          ...state.data,
+          ...action.data,
+        },
+      };
+
+    default:
+      return state;
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```

Comment:
I *expect* that now that you have typed this as a `React.FC`, you don't need this ESLint exception anymore.